### PR TITLE
RD-1488 Adding statuses and counter to sub service and envs

### PR DESCRIPTION
**Pull request by mabuaisha**
This PR is created in order to do the following:
1. Add sub statuses columns for both services and environments 
2. Add sub counters columns for both services and environments
3. Add support to check cyclic dependencies between deployment labels
4. Add support propagate deployment statuses in created graph
5. Add support to  propagate counters for deployment when add/remove node from grap

### CODE REVIEW

File: rest-service/manager_rest/rest/rest_utils.py [hunk 1]
```diff
@@ -575,6 +591,171 @@ def retrieve_and_display_dependencies(self, target_id,
...
+        :rtype Dict
+        """
+        if self.graph:
+            return
+        dependencies = self.sm.list(models.DeploymentLabelsDependencies)
+        self.graph = {}
+        for dependency in dependencies:
+            source = dependency.source_deployment_id
+            target = dependency.target_deployment_id
+            if target and target:
```
Comment (YoniItzhak) [inline rest-service/manager_rest/rest/rest_utils.py:613]: `if source and target`?

File: rest-service/manager_rest/rest/rest_utils.py [hunk 2]
```diff
@@ -575,6 +591,171 @@ def retrieve_and_display_dependencies(self, target_id,
...
 
+class RecursiveDeploymentLabelsDependencies(BaseDeploymentDependencies):
+    cyclic_error_message = 'Deployments adding labels results in ' \
+                           'cyclic deployment-labels dependencies.'
+
+    def create_dependencies_graph(self):
+        """
+        Create deployment labels dependencies from the
+        `DeploymentLabelsDependencies` model where it contains dict for all
+        holds all the graph
```
Comment (YoniItzhak) [inline rest-service/manager_rest/rest/rest_utils.py:602]: Without `holds all`?

File: rest-service/manager_rest/storage/resource_models.py [hunk 3]
```diff
@@ -456,29 +470,128 @@ def latest_execution_status(self):
...
+        return deployment_status
+
+    @property
+    def is_environment(self):
+        target_key = 'csys-obj-type'
+        target_value = 'environment'
+        for label in self.labels:
+            if label.key == target_key and label.value.lower() == target_value:
+                return True
+        else:
```
Comment (YoniItzhak) [inline rest-service/manager_rest/storage/resource_models.py:593]: No need for the `else` here

File: rest-service/manager_rest/rest/rest_utils.py (hunk 1 above)
Comment (mabuaisha) [reply rest-service/manager_rest/rest/rest_utils.py:613]: yes this should be source and target

File: rest-service/manager_rest/storage/resource_models.py (hunk 3 above)
Comment (mabuaisha) [reply rest-service/manager_rest/storage/resource_models.py:593]: yeah no need for this

**Pull request merged**